libcrux-ml-kem/src/ind_cca/incremental.rs

```diff
@@ -157,6 +157,58 @@ pub mod types {
         pub(super) randomness: [u8; 32],
     }
 
+    impl<const K: usize, Vector: Operations> EncapsState<K, Vector> {
+        /// Get the number of bytes, required for the state.
+        pub const fn num_bytes() -> usize {
+            SHARED_SECRET_SIZE
+                + vec_len_bytes::<K, Vector>()
+                + PolynomialRingElement::<Vector>::num_bytes()
+                + 32
+        }
+
+        /// Get the state as bytes
+        pub fn to_bytes(self, out: &mut [u8]) {
+            debug_assert!(out.len() >= Self::num_bytes());
+
+            out[..SHARED_SECRET_SIZE].copy_from_slice(&self.shared_secret);
+            let mut offset = SHARED_SECRET_SIZE;
+
+            vec_to_bytes(&self.r_as_ntt, &mut out[offset..]);
+            offset += vec_len_bytes::<K, Vector>();
+
+            self.error2.to_bytes(&mut out[offset..]);
+            offset += PolynomialRingElement::<Vector>::num_bytes();
+
+            out[offset..offset + 32].copy_from_slice(&self.randomness);
+        }
+
+        /// Build a state from bytes
+        pub fn from_bytes(bytes: &[u8]) -> Self {
+            debug_assert!(bytes.len() >= Self::num_bytes());
```

Comment:
This particular assert is probably best as a Result, if there's any possibility of future serialization change.

Reply:
Yeah, the functions here still panic all over the place. I'll start cleaning things up now.